### ISSUE TITLE
Generic preference/SFT dataset export from scored shadows

### ISSUE THREAD

**Issue by gotoplanb** [gotoplanb/conduct#16]
## Summary

Make "turn routed traffic into training-ready preference data" a first-class, **task-agnostic** Conduct capability. Every task with shadow eval already produces a *scored comparison* — a primary response plus N shadow responses on the same input, each scored, each stamped with model + prompt version + sampling regime. That's the raw material for DPO/SFT fine-tuning, and it exists for `dad_joke`, `email_extraction`, and `wander_eval` identically. Today it's locked in Postgres as a byproduct; this issue makes it an export surface.

This is the generic substrate behind #15. #15 should become the *first consumer* of this capability (`task_type=wander_eval`), not a bespoke Wander pipeline. No client/domain-specific logic enters Conduct.

## What is (and isn't) Conduct's responsibility

**Is:** owning the comparison structure (primary vs shadows on one input), the *lineage that makes a comparison valid to train on* (same prompt version, same sampling regime — mix these and you train on noise), a generic scoring substrate, and the export itself.

**Isn't:** any knowledge of what a task's prompt/response *means*. Conduct treats `prompt` / `response` / `inputs` as opaque blobs. It never parses `world_state` or `episode_id`. A tenant that wants domain fields re-attached for training does that as export-time enrichment against its own store, joining keys it put in the prompt.

## Building blocks that already exist

- `Job` (primary) + `JobShadow` rows on the same `parent_job_id` — the comparison structure, from the shadow-eval feature.
- `quality_scores` (1–5 + note) on `job_metadata` / `shadow_metadata` via `eval/scoring.py`.
- `prompt_version` — `prompt_loader` resolves a `version_id` into `prompt_versions`, recorded per job.
- Sampling regime — per-task `sampling` profile (deterministic/balanced/creative) shipped in 286b6de; deterministic tasks yield *reproducible* responses.

So the DPO "primary 4+, shadow 1–2 on the same input → (chosen, rejected)" query is derivable from existing tables today. This issue productizes + hardens it.

### 1. Preference + SFT export endpoints
- `GET /datasets/preferences?task_type=X&format=dpo&min_gap=2&prompt_version=V`
  → JSONL: `{prompt, system, chosen, rejected, meta}` — pair highest- vs lowest-scored response on the same input where the score gap ≥ `min_gap`.
- `GET /datasets/sft?task_type=X&min_score=4`
  → JSONL: `{prompt, system, completion, meta}` from top-scored responses.

TRL/unsloth-shaped output. Admin-auth, read-only, streamed.

### 2. Comparability lineage on every record (the Conduct-owned value-add)
A pair is invalid if its two responses came from different prompt versions or different sampling regimes. Conduct is the only party that knows these. Every export record carries — and the query can filter/group by:
- `prompt_version`, `sampling` (+ temperature/seed), `model`, `sensitivity`, `created_at`.

Default behavior: **never pair across prompt_version or sampling-regime boundaries** unless explicitly overridden. This is the feature, not a footnote.

### 3. Multi-dimensional scores (generic mechanism, client-named semantics)
Generalize `apply_score` from a single `int` to `{dimension_name: 1-5}`. Conduct stores/exports named dimensions without knowing what they mean; the export picks which dimension drives the DPO label (`...&label_dim=correctness`). Solves "my scores are lumped together" for any tenant with zero domain knowledge. Backward compatible: a bare int maps to a default dimension.

### 4. Replay-as-data-generation
`POST /jobs/{id}/replay?models=a,b,c` re-runs the stored `(prompt, system, inputs, sampling)` against new models as shadows — fanout pointed at history. Amplifies any task's dataset without new human input; reproducible when the task is deterministic. This is the synthetic-data lever.

## Export record shape (illustrative, DPO)

```json
{
  "prompt": "<opaque task prompt>",
  "system": "<resolved system prompt>",
  "chosen": "<higher-scored response>",
  "rejected": "<lower-scored response>",
  "meta": {
    "task_type": "wander_eval",
    "input_job_id": "<parent uuid>",
    "chosen": {"model": "...", "score": 5, "source": "primary"},
    "rejected": {"model": "...", "score": 2, "source": "shadow"},
    "prompt_version": 7,
    "sampling": "deterministic",
    "score_dim": "correctness"
  }
}
```

`meta` is opaque-friendly: no task carries domain fields here; the tenant joins `input_job_id` → its own context if it wants them.

## Relationship to #15

#15 (Wander Fine-Tuning Data Flywheel) becomes a consumer:
- Its DPO mining = `GET /datasets/preferences?task_type=wander_eval`.
- Its `prompt_version` concern = capability #2 (already captured).
- Its score-decomposition need = capability #3.
- Its data-volume runway = capability #4 (replay).
- Its variability/seed ask = already shipped (sampling profile); set `wander_eval` deterministic.
- The Wander-specific Loki schema (`world_state`, `transitions`) stays on the Wander side as export-time enrichment.

## Suggested sequencing

1. **`/datasets/preferences` + `/datasets/sft` over existing tables** — prove pair quality before any new storage.
2. **Lineage filters default-on** (no cross-version / cross-regime pairing).
3. **Multi-dimensional scores** — the one schema change that materially improves label quality.
4. **Replay endpoint** — synthetic amplification.
5. Loki/Grafana only if/when export volume outgrows querying Postgres — domain-agnostic, best-effort, via Alloy.

Fine-tuning itself (QLoRA/GGUF/Modelfile) stays host-side, outside Conduct; the routing-rule swap to promote a fine-tuned model is already supported.

**Comment by gotoplanb** [comment]
Closing — the LLM-as-judge feature (#17) unblocked the substance of this issue.

#16's premise was: turn routed traffic into trustworthy preference/SFT training data, and the hard part was always the *labels* — Wander flagged the human-labeling bottleneck ("~1k examples is a long runway"). The judge removes that bottleneck. The (chosen, rejected) and scored data this export was meant to produce now **accumulates automatically, in the exact lanes the export would read**:

- **Preference pairs** → the pairwise judge writes `pairwise_verdicts` (`{against_job_id, outcome: win|loss, judge_job_id}`) on both participants — order-swap-verified to cancel position bias. That *is* the (chosen, rejected) signal.
- **Pointwise / SFT scores** → the pointwise + panel judges write `quality_scores` (`via="judge"` / `via="judge-panel"`), the same lane humans use, feeding the per-model rollup.
- **Comparability lineage** (this issue's cap. #2) was already captured — `prompt_version` (resolved version id) + the `deterministic` sampling regime — so a batch is filterable to comparable conditions.

What the judge also gives that a raw score dump wouldn't: bias-controlled labels (position-swap, self-preference exclusion, panel median) and a `disagreement` flag for the cases worth a human.

**What's NOT built:** the `/datasets/preferences` + `/datasets/sft` JSONL endpoints themselves (cap. #1) and named multi-dimensional scores (cap. #3). But those are now a thin follow-on — a query over `Job ⋈ JobShadow ⋈ quality_scores / pairwise_verdicts` shaped into TRL/unsloth JSONL — not a data-generation problem. Open a fresh issue/PR for the endpoint when you actually want to pull a dataset; the data's already there and growing.

The domain boundary held throughout: Conduct produces task-agnostic scored comparisons; tenants (Wander) reshape/enrich at export time. See #17 + docs/judging.md.
